core/src/ast/modules.rs

```diff
@@ -106,6 +106,17 @@ pub struct Module {
     pub attrs: Attrs,
 }
 
+struct ModuleSynTraverse {
```

Comment:
nit: docs please.

Probably call this "ModuleBuilder"?

And `add_item` can be a `.add(&mut self)` method on this.